pyk/src/pyk/ktool/krun.py

```diff
@@ -115,6 +118,7 @@ def run(
         pipe_stderr: bool = True,
         bug_report: BugReport | None = None,
         debugger: bool = False,
+        proof_hint: bool = False,
```

Comment:
Unfortunately, this approach won't work: under the hood it leads to a call like

```python3
subprocess.run(['krun', 'test.imp', '--proof-hint'], text=True, stdout=subprocess.PIPE)
```

which will raise an exception:

```python3
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 8: invalid start byte
```

This is because
* `stdout=subprocess.PIPE`, so `CompletedProcess.stdout` will be populated.
* `text=True`, so the function will try to decode the bytes output to produce `CompletedProcess.stdout`.

You drop any of these arguments and it'll work.

Workarounds:
1. Reuse `_build_arg_list` and call `subprocess.run` directly.
2. If we know how the bytes output is supposed to be decoded, we can thread argument `encoding` through to `subprocess.run` ([standard encodings](https://docs.python.org/3/library/codecs.html#standard-encodings)).
3. You can pass `pipe_stdout=False` to `run_process`, so that you have `stdout=None` in `subprocess.run`.
4. We can patch `run_process` (and it's newer version, `run_process_2` ) to accept `text=False`, but that's not straightforward.

I'd prefer a solution that won't prevent us from dropping `run_process` in favor of `run_process_2` in the future, so (1) or (2).

Reply:
You're right! I dropped `text=True` locally and forgot to commit this.

Reply:
About your workarounds:
1. Isn't that just a different way to create a third `run_process`?
2. I don't think we can do this as the encoding was created by un on the llvm backend serializer; we really need to output the binary itself, not its text representation/deserialization.
3. That's a good approach; we can have `pipe_stdout = not prof_hint`.
4. Indeed, that's not trivial and isn't worth spending time on. 

If you don't mind, I would prefer to stick to option 3 to unblock Pi2 folks, and when we need to drop `run_process` in favor of `run_process_2`, we can sit together again to brainstorm a way to make it compatible with proof hints.